### DISCUSSION TITLE
Composite/mixed likelihood

### DISCUSSION THREAD

**Discussion by ankushaggarwal** [cornellius-gp/gpytorch#1639] [Q&A]
I am looking to use a Gaussian likelihood for the output and a Bernouille likelihood for the derivative of the output. What is the best way to achieve this? Is there a likelihood class that I can use or would I need to implement one? If the latter approach is required, any suggestions on where to start? Thanks!

**Comment by gpleiss** [comment]
> Is there a likelihood class that I can use or would I need to implement one? If the latter approach is required, any suggestions on where to start?

You would have to implement your own custom likelihood. Also, you'd have to use a variational model due to the non-conjugacy of the likelihood - and at the moment variational models do not support derivatives. This should be something that is supported soon I think (@jacobrgardner?)

**Reply by ankushaggarwal** [reply]
Thanks @gpleiss  Any suggestions for examples to follow while implementing a custom likelihood? I have already done derivatives with variational models - it may be more of a hack and not an elegant solution, but it works.

**Reply by gpleiss** [reply]
Sorry for the slow reply @ankushaggarwal . I'd follow what's written in BetaLikelihood or LaplaceLikelihood in `gpytorch/likelihoods/`. These likelihoods do not handle derivatives, however.
